client/question/question.js

```diff
@@ -550,101 +550,195 @@ tie.constant(
   'PREREQ_CHECK_TYPE_INVALID_STUDENTCODE_CALL', 'invalidStudentCode');
 
 /**
-  Dictionary of the customized runtime error feedback messages
-  - `checker` function checks to see if the error matches with the type of
-    feedback message
-  - `generateMessage` function returns the appropriate feedback message for the
-    given error
+ * Dictionary of the novice-friendly syntax error feedback messages
+ *
+ * `friendlyErrorCheck` function checks to see if the error matches with the
+ * type of feedback message
+ *
+ * `getFriendlyErrorText` function returns the appropriate feedback message for
+ * the given error
+ *
+ * @type {Object}
+ */
+tie.constant('FRIENDLY_SYNTAX_ERROR_TRANSLATIONS', {
+  python: [
+    {
+      friendlyErrorCheck: function(errorString) {
+        var re = new RegExp('^NameError: global name .* is not defined$');
+        return re.test(errorString);
+      },
+      getFriendlyErrorText: function(errorString) {
+        var re = new RegExp('NameError: global name (.*) is not defined');
+        var varName = errorString.match(re)[1];
+        return [
+          '<code>', varName, '</code> appears to be a name that was not ',
+          'previously declared or defined. E.g., if <code>', varName,
+          '</code> is a variable to hold a string, <code>', varName,
+          ' = \'\'</code> would declare <code>', varName, '</code> as a ',
+          'string variable. Similarly, <code>', varName, ' = []</code> ',
+          'declares <code>', varName, '</code> as an array variable. Another ',
+          'possibility is that <code>', varName, '</code> was misspelled or ',
+          'uses incorrect capitalization.'].join('');
+      }
+    }, {
+      friendlyErrorCheck: function(errorString) {
+        var re = new RegExp('^SyntaxError: invalid syntax');
+        return re.test(errorString);
+      },
+      getFriendlyErrorText: function() {
+        return [
+          'This is the default, generic syntax error message in Python. Here ',
+          'are some common mistakes that can result in this error:',
+          '<ul>',
+          '<li>Mis-matching (missing or too many) braces, brackets, ',
```

Comment:
The closing li tags are all missing.

Reply:
Some style guides say this is preferable (shrug). I can add the closing tags if you prefer though.

Reply:
Hm, interesting. Yes, I would prefer having the closing tags, if that's OK.

Reply:
OK. Fixed in next commit.